tools/caml_tex.ml

```diff
@@ -134,8 +134,17 @@ module Toplevel = struct
 
   (** Record the main location instead of printing it *)
   let printer_register_locs =
-    { Location.batch_mode_printer with
-      pp_main_loc = (fun _ _ _ loc -> register_loc loc) }
+    let p = Location.batch_mode_printer in
+    Location.{
+      pp = p.pp;
+      pp_report_kind = p.pp_report_kind;
+      pp_main_loc = (fun _ _ _ loc -> register_loc loc);
+      pp_main_txt = p.pp_main_txt;
+      pp_submsgs = p.pp_submsgs;
+      pp_submsg = p.pp_submsg;
+      pp_submsg_loc = (fun _ _ _ loc -> register_loc loc);
+      pp_submsg_txt = p.pp_submsg_txt;
+    }
```

Comment:
This seems to be the right fix but why not use

```OCaml
{ Location.batch_mode_printer with
   pp_main_loc = (fun _ _ _ loc -> register_loc loc);
   pp_submsg_loc = (fun _ _ _ loc -> register_loc loc);
}
```
?

Reply:
This is to avoid the same bug again. I agree it is less readable, I reverted.

Reply:
I was the one who suggested to @Julow to list the fields in full. The reason is that I want to make sure that we revisit this place whenever we add new fields, to make sure that we don't silently introduce a bug again, as @Armael did when he introduced `pp_submsg_loc` in the field place. (Think of the dual of fragile pattern matches.)

However, @Armael and @Octachron both dislike this fully-expanded version (I tried to convince @Armael just now, but failed). I believe they are both wrong, but in the interest of getting the work done I suggest moving forward with the (wrong) version that they like better.

Reply:
I am not opposed to the fully expanded version, I was just surprised to see this change between the first and second version of the commit. In term of readability, reorganizing the fields to have
first the updated fields and then fields here for future-proofing, e.g
```OCaml
{ 
   pp_main_loc = (fun _ _ _ loc -> register_loc loc);
   pp_submsg_loc = (fun _ _ _ loc -> register_loc loc);

  (* The following fields are kept identical to [p] but explicited to catch future changes *)
  pp_submsg = p.submsg
  ...
}
```
might be a good compromise?

Reply:
I'll just merge the PR, we can improve this later if we want to. (You blew your chance!)